app/app/src/main/jni/vsp-pty.cpp

```diff
@@ -195,79 +190,53 @@ static void* ptyThread(void* irrelevant)
         {
             //            buf[nread] = '\0';
 
-            if (CHECK_BIT(*buf, TIOCPKT_DATA))
-            {
+            if (*buf == TIOCPKT_DATA)
                 __android_log_print(ANDROID_LOG_VERBOSE, "TAG", "TIOCPKT_DATA\n");
-            }
 
-            if (CHECK_BIT(*buf, TIOCPKT_START))
-            {
+            if (*buf == TIOCPKT_START)
                 __android_log_print(ANDROID_LOG_VERBOSE, "TAG", "TIOCPKT_START\n");
-            }
 
-            if (CHECK_BIT(*buf, TIOCPKT_STOP))
-            {
+            if (*buf == TIOCPKT_STOP)
                 __android_log_print(ANDROID_LOG_VERBOSE, "TAG", "TIOCPKT_STOP\n");
-            }
 
-            if (CHECK_BIT(*buf, TIOCPKT_IOCTL))
-            {
+            if (*buf == TIOCPKT_IOCTL)
                 __android_log_print(ANDROID_LOG_VERBOSE, "TAG", "IOCTL\n");
-            }
 
-            if (CHECK_BIT(*buf, TIOCPKT_FLUSHREAD))
-            {
+            if (*buf == TIOCPKT_FLUSHREAD)
                 __android_log_print(ANDROID_LOG_VERBOSE, "TAG", "TIOCPKT_FLUSHREAD\n");
-            }
 
-            if (CHECK_BIT(*buf, TIOCPKT_FLUSHWRITE))
-            {
+            if (*buf == TIOCPKT_FLUSHWRITE)
                 __android_log_print(ANDROID_LOG_VERBOSE, "TAG", "TIOCPKT_FLUSHWRITE\n");
-            }
 
-            if (CHECK_BIT(*buf, TIOCPKT_DOSTOP))
-            {
+            if (*buf == TIOCPKT_DOSTOP)
                 __android_log_print(ANDROID_LOG_VERBOSE, "TAG", "TIOCPKT_DOSTOP\n");
-            }
 
-            if (CHECK_BIT(*buf, TIOCPKT_NOSTOP))
-            {
+            if (*buf == TIOCPKT_NOSTOP)
                 __android_log_print(ANDROID_LOG_VERBOSE, "TAG", "TIOCPKT_NOSTOP\n");
-            }
 
-            struct termios tio;
+            struct termios tio={};
             tcgetattr(master, &tio);
             __android_log_print(ANDROID_LOG_VERBOSE, "TAG", "Baudrate: %d\n", cfgetospeed(&tio));
-            if (tio.c_cflag & CS8)
```

Comment:
These CS* bit masks actually must be filtered with CSIZE before comparison

Reply:
Actually most of those calls do nothing right now. I should probably do a larger refactor here. The idea was to get parity, stop bits, and bit size parameters from here and pass them through JNI, but for now I'm just passing the raw values of termios inside the `SerialData` object.